src/source/libscalene.cpp

```diff
@@ -141,6 +141,7 @@ class MakeLocalAllocator {
   }
 
   static inline void *local_malloc(void *ctx, size_t len) {
+    MallocRecursionGuard m;
```

Comment:
Recursion guard added here because we only want memory accounting at the top level-- if it's passed from `local_malloc` into the system malloc, we still want to attribute it here.